Make iteration results iterable

ljharb opened the pull request:
This allows, for example, `const [firstPackage] = require('find-package-json')(dir);`, because the iteration result will now itself be iterable. This matches the convention of all built-in iterators in the language.

3rd-Eden commented:
Is it possible that you add a test for this to ensure that the suggest functionality does not break in future updates?

ljharb commented:
Sure, happy to!

ljharb committed:
Make iteration results iterable

This allows, for example, `const [firstPackage] = require('find-package-json')(dir);`, because the iteration result will now itself be iterable. This matches the convention of all built-in iterators in the language.

ljharb commented:
Updated.

3rd-Eden commented:
I will release a new minor version in a few (as this adds a new feature).

3rd-Eden commented:
Thanks again for taking the time to contribute.
